logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/Message.kt

```diff
@@ -341,8 +342,27 @@ sealed interface Message {
         }
     }
 
-    enum class Status {
-        PENDING, SENT, READ, FAILED, FAILED_REMOTELY
+    sealed class Status {
+
+        object Pending : Status()
+        object Sent : Status()
+
+        object Delivered : Status()
+
+        data class Read(val readCount: Long) : Status()
+
+        object Failed : Status()
+
+        object FailedRemotely : Status()
+
+        override fun toString(): String = when (this) {
```

Comment:
Can't we already use `data object`? Then we wouldn't need to override `toString` 🤔

Reply:
with the change proposed by Tommaso we will get the logs in json format with the corresponding keys

Reply:
Yup, right, I opened the changes from when Tommaso wrote these comments, without further changes, now I can see the whole picture